### PR TITLE
[feature] Docker

### PR DESCRIPTION
**Pull request by TheGamer1973**
## Status
**READY**

## Migrations
NO

## Description
Creating Dockerfile and Github Templates for the project.

### CODE REVIEW

**Review by Eduardoveras — APPROVED** [review]

**Review by yimialmonte — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM node:7
+
+# Establishing Working Directory
+WORKDIR /usr/src/app
+
+# Copying Files into Directory
+COPY . .
+
+# Updating Systems and Adding PostgreSql Repo
+RUN npm install --quiet --global \ vue-cli && apt-get update -qq && apt-get install git -y -V
```
Comment (jsainthilaire) [inline Dockerfile:10]: why are we running npm install twice?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
+FROM node:7
+
+# Establishing Working Directory
+WORKDIR /usr/src/app
```
Comment (jsainthilaire) [inline Dockerfile]: we could use an app or src directly because mounting a folder structure following /usr will be confusing

**Comment by jsainthilaire** [comment]
@TheGamer1973 why don't we just build and push the docker image so people just need to pull an already created image, this will help with consistency and save a step in the dev process?

**Review by jsainthilaire — CHANGES_REQUESTED** [review]
@TheGamer1973 why don't we just build and push the docker image so people just need to pull an already created image, this will help with consistency and save a step in the dev process?

**Comment by jsainthilaire** [comment]
also before merging you should squash and clean these commits.

**Comment by TheGamer1973** [comment]
As for the Squash, after approving the commit Github gives the possibility to Squash and Merge. Should I use that approach or should I do it before?

**Comment by TheGamer1973** [comment]
Made some changes as requested.

**Comment by jsainthilaire** [comment]
I will approve this for now, but the process needs to be rethought.

**Review by jsainthilaire — APPROVED** [review]

**Pull request merged**